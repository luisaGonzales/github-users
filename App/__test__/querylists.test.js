import React from 'react';
import renderer from 'react-test-renderer';
import UserList from '../Components/UsersListQuery';
import RepoList from '../Components/ReposListQuery';
import 'isomorphic-fetch';
import { InMemoryCache } from 'apollo-cache-inmemory';
import ApolloClient from "apollo-boost";
import token from '../../config';
import { ApolloProvider } from 'react-apollo';

const client = new ApolloClient(
  {
    uri: 'https://api.github.com/graphql',
    headers: {
      authorization: `Bearer ${token}`
    },
    cache: new InMemoryCache({
      dataIdFromObject: obj => obj.id,
      addTypename: false,
      fragmentMatcher: {
        match: ({ id }, typeCond, context) => !!context.store.get(id)
      }
    })
  }
  
);

test('User list query', async () => {
  const userList = renderer.create(
    <ApolloProvider client={ client } >  
      <UserList user={"Luisa"}/>
    </ ApolloProvider>
  ).toJSON();

  expect(userList).toMatchSnapshot()
})

test('Repo list query', async () => {
  const userList = renderer.create(
    <ApolloProvider client={ client } >  
      <RepoList login={"luisaGonzales"}/>
    </ ApolloProvider>
  ).toJSON();

  expect(userList).toMatchSnapshot()
});