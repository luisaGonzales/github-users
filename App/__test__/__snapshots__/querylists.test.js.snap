// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo list query 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#333"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`RepoList query without error 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#333"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`User list query 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={<Styled(ViewNB) />}
    ListFooterComponent={
      <ActivityIndicator
        animating={true}
        color="#333"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
          }
        }
      />
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {},
            undefined,
            null,
          ]
        }
      />
      <View
        onLayout={[Function]}
      >
        <ActivityIndicator
          animating={true}
          color="#333"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`UserList query without error 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={<Styled(ViewNB) />}
    ListFooterComponent={
      <ActivityIndicator
        animating={true}
        color="#333"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
          }
        }
      />
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {},
            undefined,
            null,
          ]
        }
      />
      <View
        onLayout={[Function]}
      >
        <ActivityIndicator
          animating={true}
          color="#333"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
